packages/valory/skills/decision_maker_abci/behaviours/sell_outcome_token.py

```diff
@@ -0,0 +1,181 @@
+
+from packages.valory.contracts.market_maker.contract import (
+    FixedProductMarketMakerContract,
+)
+from packages.valory.protocols.contract_api import ContractApiMessage
+from packages.valory.skills.decision_maker_abci.behaviours.base import (
+    DecisionMakerBaseBehaviour,
+    WaitableConditionType,
+)
+from packages.valory.skills.decision_maker_abci.models import MultisendBatch
+from packages.valory.skills.decision_maker_abci.payloads import MultisigTxPayload
+from packages.valory.skills.decision_maker_abci.states.sell_outcome_token import (
+    SellOutcomeTokenRound,
+)
+
+
+class SellTokenBehaviour(DecisionMakerBaseBehaviour):
+    """A behaviour in which the agents sell a token."""
+
+    matching_round = SellOutcomeTokenRound
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the sell token behaviour."""
+        super().__init__(**kwargs)
+        self.sell_amount: float = 0.0
+        self.return_amount: int = 0
+
+    @property
+    def market_maker_contract_address(self) -> str:
+        """Get the contract address of the market maker on which the service is going to place the bet."""
+        return self.sampled_bet.id
+
+    @property
+    def outcome_index(self) -> int:
+        """Get the index of the outcome for which the service is going to sell token."""
+        return cast(int, self.synchronized_data.vote)
+
+    def _build_approval_tx(self) -> WaitableConditionType:
+        """Build an ERC20 approve transaction."""
+        approval_tx = yield from self.build_approval_tx(
+            self.return_amount,
+            self.market_maker_contract_address,
+            self.market_maker_contract_address,
+        )
+        return approval_tx
+
+    def _calc_sell_amount(self) -> WaitableConditionType:
+        """Calculate the sell amount of the conditional token."""
+        response_msg = yield from self.get_contract_api_response(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            contract_address=self.market_maker_contract_address,
+            contract_id=str(FixedProductMarketMakerContract.contract_id),
+            contract_callable="calc_sell_amount",
+            return_amount=self.return_amount,
+            outcome_index=self.outcome_index,
+        )
+        if response_msg.performative != ContractApiMessage.Performative.RAW_TRANSACTION:
+            self.context.logger.error(
+                f"Could not calculate the sell amount: {response_msg}"
+            )
+            return False
+
+        sell_amount = response_msg.raw_transaction.body.get(
+            "outcomeTokenSellAmount", None
+        )
+        if sell_amount is None:
+            self.context.logger.error(
+                f"Something went wrong while trying to get the outcomeTokenSellAmount amount for the conditional token: {response_msg}"
+            )
+            return False
+
+        self.sell_amount = sell_amount
+        return True
```

Comment:
This has a lot of code dup with the calc buy method, therefore we could create a generic calc amount method with a flag `buy` and calc the buy or sell amount based on the flag.